crates/cubecl-core/src/pod.rs

```diff
@@ -1,17 +1,21 @@
+use bytemuck::NoUninit;
+
 use crate::{
     flex32,
     ir::{Elem, FloatKind, IntKind, UIntKind},
     prelude::Numeric,
 };
 
 /// The base element trait for the jit backend.
-pub trait CubeElement: core::fmt::Debug + Send + Sync + 'static + Clone + bytemuck::Pod {
+pub trait CubeElement: core::fmt::Debug + Send + Sync + 'static + Clone + NoUninit {
```

Comment:
I don't think it's wise to remove `bytemock::Pod`. Some primitive types just can't be loaded into global memory like `bools`. And I don't think we want to `force` a representation for booleans. At some point we might simply encode 32 bools into a single `u32` to save space and using masking to retrieve the right value, but that would require padding. And that part should be done in Burn, like quantization.